association.go

```diff
@@ -384,18 +450,118 @@ func (a *Association) handleInbound(raw []byte) error {
 	}
 
 	for _, c := range p.chunks {
-		packets, err := a.handleChunk(p, c)
+		err := a.handleChunk(p, c)
 		if err != nil {
 			return errors.Wrap(err, "failed handling chunk")
 		}
-		for _, p := range packets {
-			err = a.send(p)
-			if err != nil {
-				return errors.Wrap(err, "failed sending reply")
+	}
+
+	return nil
+}
+
+// handleOutbound handles outgoing processes
+func (a *Association) handleOutbound() error {
+	a.lock.Lock()
+
+	packets := []*packet{}
+	if a.controlQueue.size() > 0 {
+		packets = append(packets, a.controlQueue.popAll()...)
+	}
+
+	state := a.getState()
+
+	if state == established {
+		if !a.willRetransmitDataChunks {
```

Comment:
tiny suggestion: IMHO I think inverting the conditions to have the positive case first will make it more readable

Reply:
OK.